### PR TITLE
Switch fixes

### PR DESCRIPTION
**Pull request by l-Luna**
Fixes
- Switch statements with `null` labels being decompiled as type labels
- Switch expressions with `null` values being decompiled as the name of the expression type
- Nullable enum switch resugaring
- Type switches with only one code block not being resugared
- Type switches leaving behind an unused synthetic variable

### CODE REVIEW

File: src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java [hunk 1]
```diff
@@ -318,6 +341,29 @@ private static boolean isEnumArray(Exprent exprent) {
   private static ArrayExprent getEnumArrayExprent(Exprent switchHead, RootStatement root) {
     Exprent candidate = switchHead;
 
+    if (switchHead instanceof FunctionExprent) {
```
[the remark this answers is not in the document]
Comment (l-Luna) [reply src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java:344]: it tests that the switch head is a ternary of the right shape, and that the true branch follows the other criteria for a switch map, just below

[the remark this answers is not in the document]
Comment (SuperCoder7979) [reply src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java:344]: Right, yeah. I was wondering what happens when you have, for example:
```java
static int testTriangle(Direction d, Direction d2, boolean b) {
    return switch (b ? d : d2) {
      case NORTH, SOUTH, WEST -> 0;
      case EAST, UP -> 1;
      case DOWN, null, default -> -1;
    };
  }
```

[the remark this answers is not in the document]
Comment (l-Luna) [reply src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java:344]: works fine, the ternary becomes another variable

[the remark this answers is not in the document]
Comment (SuperCoder7979) [reply src/org/jetbrains/java/decompiler/modules/decompiler/SwitchHelper.java:344]: neat, that clears it up

**Review by SuperCoder7979 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**